test/functional/api/cas/cli_messages.py

```diff
@@ -252,23 +252,48 @@
     r"Option '--cache-line-size \(-x\)' is not allowed"
 ]
 
-def check_stderr_msg(output: Output, expected_messages, negate=False):
-    return __check_string_msg(output.stderr, expected_messages, negate)
+partition_not_suitable_for_array = [
+    r'\S+ is not suitable for this array'
+]
+
+device_or_resource_busy = [
+    r"cannot open \S+: Device or resource busy"
+]
+
+
+def check_stderr_msg_all(output: Output, expected_messages):
+    return __check_string_msg_all(output.stderr, expected_messages)
+
+
+def check_stdout_msg_all(output: Output, expected_messages):
+    return __check_string_msg_all(output.stdout, expected_messages)
+
 
+def check_stderr_msg_any(output: Output, expected_messages):
+    return __check_string_msg_any(output.stderr, expected_messages)
 
-def check_stdout_msg(output: Output, expected_messages, negate=False):
-    return __check_string_msg(output.stdout, expected_messages, negate)
 
+def check_stdout_msg_any(output: Output, expected_messages):
+    return __check_string_msg_any(output.stdout, expected_messages)
 
-def __check_string_msg(text: str, expected_messages, negate=False):
+
+def __check_string_msg_all(text: str, expected_messages):
     msg_ok = True
     for msg in expected_messages:
         matches = re.search(msg, text)
-        if not matches and not negate:
+        if not matches:
             TestRun.LOGGER.error(f"Message is incorrect, expected: {msg}\n actual: {text}.")
             msg_ok = False
-        elif matches and negate:
-            TestRun.LOGGER.error(f"Message is incorrect, expected to not find: {msg}\n "
-                                 f"actual: {text}.")
-            msg_ok = False
+    return msg_ok
+
+
+def __check_string_msg_any(text: str, expected_messages):
+    msg_ok = False
+    for msg in expected_messages:
+        matches = re.search(msg, text)
+        if matches:
+            msg_ok = True
+            break
+    if not msg_ok:
+        TestRun.LOGGER.error(f"Message is incorrect: {text}.")
```

Comment:
How about only one function but with an optional parameter `match_all_messages`?
```
def __check_string_msg(text: str, exptected_messages, match_all_messages=True):
    matches = [m for m in expected_messages if re.search(m, text)]
    if not match_all_messages and matches:
        return True
    elif match_all_messages and set(matches) == set(expected_messages):
        return True

    TestRun.LOGGER.error(f"Message is incorrect, expected: '{msg}'\n actual: '{text}'.")
    return False
````
This way you can avoid the huge refactor.